### ISSUE TITLE
Segment Fault because recursive function calling.

### ISSUE THREAD

**Issue by lohikaarme** [Andrettin/Wyrmgus#125]
I was making 512×512 map which had rescue-active and passive **towns** in it.
When I try to test run the map segment fault happened while loading forest.png.

I think this picture of gdb's where command tells the tale.
![toomanyfunctioncalls](https://user-images.githubusercontent.com/2855916/40905526-e1753504-67e6-11e8-994b-72ca056fffbc.png)

Best solution to me sound like removing the recursion. Which sound like fun project after I'm done at work.

**Comment by Andrettin** [comment]
Thanks for the bug report!

Did the map have settlement sites owned by players other than the neutral player?

If so, this commit could fix the issue:
https://github.com/Andrettin/Wyrmgus/commit/c2b6412a6ae43d485de7fc615aca9e14e8365b4f

**Comment by lohikaarme** [comment]
There was empty settlement site. (I needed "stone gathering place" for Goblins). According to sms file faction was set to 13 (rescue faction).
After running I did get Lua error message about first parameter of GetCivilizationData being nil (upgrade-functions.lua:68). However it seems to be related to custom dwarf faction I made not to settlement site. (I had backup copy where settlement wasn't place yet.) So I think we can say it is unrelated bug. I close this now and after debugging more make other ticket.... to other repository since it is Lua related?
